packages/styleguide/src/components/Discussion/Tree/CommentNode.tsx

```diff
@@ -270,6 +270,15 @@ const CommentNode = ({
   )
 
   if (isExpanded) {
+    const { handleUpVote, handleDownVote, handleUnVote } = actions
+    const voteActions = handleUpVote &&
+      handleDownVote &&
+      handleUnVote && {
+        handleUpVote,
+        handleDownVote,
+        handleUnVote
+      }
```

Comment:
I am not sure if this is the best way to handle this. TypeScript is requiring all these handlers to be present. I suppose the best way would be to make them optional on the CommentNode and handle it with that. (With this implementation, the Project could no longer be built if TS-Errors block the build)

Reply:
+1 
The above fix is a bit confusing – plus then the type structure doesn't reflect the comments' reality anymore. How complex would it be to nest vote actions in actions and make them optional as a whole, but individually mandatory if present?
```
type Actions = {
  ...
  voteActions?: {
     handleUpVote: Fn
     handleDownVote: Fn
     handleUnVote: Fn
  }
}
```

Reply:
Addressed in https://github.com/republik/plattform/pull/3/commits/6f4a604684f2f5db75920aa8ea5c58d803307378 and going for dedicated `voteActions`.